Add audio context file

KageJordan opened the pull request.

KageJordan committed:
Add audio context file

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,151 @@
+  <title>Web Audio Api - Audio Context</title>
+  
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
```
Drop space in value.  Then can drop quotes 😎

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,151 @@
+      --line-height: 4em;
+    }
+    
+    @media(min-width: 250px) {
```
🤩

KageJordan replied:
👍🏽

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,151 @@
+ 
+<fieldset><legend>Sweep</legend>
+  <label>Attack
+    <input type="range"name="Attack" min="0" max="1" value="0.2" step="0.1">
```
Attributes missing space separation. Now this WILL confuse parser says quotes

KageJordan replied:
👍🏽

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,151 @@
+  </label>
+
+  <fieldset><legend hidden>Checkboxes</legend>
+    <label><input name="Pad1" type="checkbox"></label>
```
`name`s for checkboxes +radio  buttons should be the same. This is how html links value(s) to key in form submission.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,151 @@
+ 
+<fieldset><legend>Pulse</legend>
+  <label>Hz
+    <input type="range" name="Hertz" min="660" max="1320" value="880" step="1">
```
Iowercase name.  Save the SHIFT press 😉

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,151 @@
+  </label>
+ 
+  <fieldset><legend hidden>Checkboxes</legend>
+    <label><input name="Pad5" type="checkbox"></label>
```
See above in first fieldset

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,151 @@
+  </label>
+ 
+  <fieldset><legend hidden>Checkboxes</legend>
+    <label><input name="Pad9" type="checkbox"></label>
```
See above in first fieldset

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,151 @@
+  </label>
+
+  <fieldset><legend hidden>Checkboxes</legend> 
+    <label><input name="Pad13" type="checkbox"></label>
```
See above in first fieldset

snuggs reviewed (changes requested):
A few nits. Looking good.

snuggs committed:
Add behavior configuration  to <form> events

Adds bindings for `submit`, `formdata`, `change`, & `input` form events. Perhaps missing `cancel`?

snuggs commented:
@KageJordan please review :point_right: https://github.com/KageJordan/web-audio/pull/1/commits/fd4422f64cb9a94bb72ce1941163007ef2ef514f

KageJordan committed:
Update audio_context.html

KageJordan committed:
Update audio_context.html

KageJordan committed:
Update audio_context.html

KageJordan committed:
Update audio_context.html

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,321 @@
+<body>
+
+  <header>
+    <h1>DevPunks Sequencer</h1>
```
Headers on headers? Seems like duplicated content w form header and masthead.  🤔

KageJordan replied:
One of the headers was for the masthead of the page and the other was for the sequencer, I could change it to another element like an H2 but I remember us talking about having a header & two nested <label> tags with nested input tags inside of them.I could also change the name of the masthead to something else ?

snuggs replied:
see below...

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,321 @@
+
+    <label>
+      BPM
+      <input type=range min=10 max=522 value=251>
```
What are your thoughts on this order for attributes.
```html
<input name=... value=... min=... max=... type=range >
```
Keeps the human interested stuff up front starting wirh `input` variable `name=...` and `value=...`.  The `type=...` is more an implementation detail hence last in order. Can even leave it off for `type=text` since that is the default for `<input ...>`. Just a thought...

KageJordan replied:
I prefer this order, will change it around.

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,321 @@
+
+    <fieldset>
+
+      <legend hidden>Checkboxes</legend>
```
When you figure out a better legend text you want to keep that content by [making it _"invisible"_ not `hidden` _(read comments)_](https://stackoverflow.com/questions/6708247/what-is-the-difference-between-the-hidden-attribute-html5-and-the-displaynone). Fine for now to yank it out the dom but remember for later.  Ultimately there should be no `hidden`s unless controlled by javascript.  But fine to use during development. Just remember to remove before going live, like `console.log`s.

KageJordan replied:
Sounds good ,I hadn't thought about using display:none.Gracias

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,321 @@
+      <legend hidden>Checkboxes</legend>
+
+      <label>
+        <input name=pads type="checkbox">
```
Interesting if you can make a convention around the `value=...`. Perhaps some convention like `value=thunder` then can use js to retrieve value and load `thunder.mp3` or `new ThunderCompressor` etc.

KageJordan replied:
👍🏽

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,321 @@
+      <input name=pads type="checkbox">
+    </label>
+
+  </fieldset>
```
Triple closing `</fieldset>`s on the same nesting level? 🤔

KageJordan replied:
It should be one < fieldset> with 4 nesting < fieldset>s within , might have created too many spaces between tags, could be confusing.

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,321 @@
+
+      <label>
+        Rate
+      <input type="range" name="rate" min="0.1" max="2" value="1" step="0.1">
```
One more indent.

KageJordan replied:
👍🏽

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,321 @@
+      form.addEventListener ('submit', submit)
+
+    for (let fieldset of fieldsets)
+    configure(fieldset)
```
One more indent.  `for` loops typically have curly braces but if you only have one line of code you can forego curly braces.  Typically indent that one line 1 level deeper than the `for` loop for readability.

KageJordan replied:
👍🏽

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,321 @@
+
+    for (let range of ranges) {
+    // https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/input_event
+    // https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener
```
Code inside block nesting is flat.

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,321 @@
+      const ranges = fieldset.querySelectorAll ('[type=range]')
+        , checkboxes = fieldset.querySelectorAll ('[type=checkbox]')
+
+    for (let range of ranges) {
```
`for` loop same nesting level as parent  function.

snuggs reviewed (changes requested):
A few nits

KageJordan committed:
<refactor>:add indentation, fix input elements

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,321 @@
+
+  main {
+    outline: thick double;
+    background-color: whitesmoke;
```
...as opposed to _"WHITE SNAKE"_ 🐍😃

KageJordan replied:
I see what you did there 😺

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,321 @@
+  <form action=#>
+
+  <header>
+    DevPunks Sequencer
```
@KageJordan this is what I mean. Seems like there is a `<h1>` with the same content above.

KageJordan replied:
Will remove confusing  content, has it is not very accessible.

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,321 @@
+
+    <label>
+      Hz
+      <input type="range" name="hertz" min="660" max="1320" value="880" step="1">
```
May want to give same `name=... value=...` treatment as the rest. No need to change now. technically _zero_  "wrong" with this code. Well done!

KageJordan replied:
thanks for pointing it out  ,need to stay consistent with all < input> elements, changing order around

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,321 @@
+
+    <label>
+      LFO
+      <input type="range" name="lfo" min="20" max="40" value="30" step="1">
```
Same as above comment. _(rest seem consistent order)_.

snuggs reviewed (approved):
LGTM! Let's discuss next commits to see what's left for this PR/branch before merging. :trophy:

KageJordan committed:
<refactor>:remove content ,reorder content

Removed content from second <header >
Reordered content in <input > elements

snuggs commented on audio_context.html:
```diff
@@ -0,0 +1,320 @@
+<body>
+
+  <header>
+   <h1>DevPunks Sequencer</h1>
```
Off by 1 (whitespace) error? 😉

snuggs reviewed (commented):
Pretty good. Basically splitting html hairs at this point.

snuggs committed:
-mAdd bpm handler & play handler
